crates/interface-types/src/lib.rs

```diff
@@ -141,12 +142,55 @@ impl ModuleData {
         let incoming = binding.param_bindings()?;
         let outgoing = binding.result_bindings()?;
 
-        let wasm_args = translate_incoming(cx, handle, &incoming, args)?;
+        let wasm_args =
+            translate_incoming(&mut RawTranslateContext::new(cx, handle), &incoming, args)?;
         let wasm_results = match cx.invoke(handle, export, &wasm_args)? {
             ActionOutcome::Returned { values } => values,
             ActionOutcome::Trapped { message } => bail!("trapped: {}", message),
         };
-        translate_outgoing(cx, handle, &outgoing, &wasm_results)
+        translate_outgoing(
+            &mut RawTranslateContext::new(cx, handle),
+            &outgoing,
+            &wasm_results,
+        )
+    }
+
+    /// Invokes wasmtime function with a `&[Value]` list. `Value` the set of
+    /// wasm interface types.
+    pub fn invoke_export(
+        &self,
+        instance: &api::HostRef<api::Instance>,
+        export: &str,
+        args: &[Value],
+    ) -> Result<Vec<Value>> {
+        let mut handle = instance.borrow().handle().clone();
+
+        let binding = self.binding_for_export(&mut handle, export)?;
+        let incoming = binding.param_bindings()?;
+        let outgoing = binding.result_bindings()?;
+
+        let f = instance
+            .borrow()
+            .find_export_by_name(export)
+            .ok_or_else(|| format_err!("failed to find export `{}`", export))?
+            .func()
+            .ok_or_else(|| format_err!("`{}` is not a function", export))?
+            .clone();
+
+        let mut cx = InstanceTranslateContext(instance.clone());
+        let wasm_args = translate_incoming(&mut cx, &incoming, args)?
+            .into_iter()
+            .map(|rv| rv.into())
+            .collect::<Vec<_>>();
+        let wasm_results = match f.borrow().call(&wasm_args) {
+            Ok(values) => values,
+            Err(trap) => bail!("trapped: {:?}", trap),
+        }
+        .to_vec()
+        .into_iter()
+        .map(|v: api::Val| v.into())
+        .collect::<Vec<RuntimeValue>>();
```

Comment:
Could a temporary be introduced here? The usage of `match` as an expression leads to some weird formatting